### PR TITLE
must-gather: remove mustgather scripts

### PR DESCRIPTION
**Pull request by yati1998**
This commit removed must-gather scripts
and other related scripts and docs from the
repo as it is now added to odf-must-gather

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

**Comment by yati1998** [comment]
@agarwal-mudit can you please review this PR.

File: Makefile [hunk 1]
```diff
@@ -11,11 +11,10 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
-FUSION?=false
```
Comment (malayparida2000) [inline Makefile:14]: This shouldn't be removed

File: Makefile [hunk 2]
```diff
@@ -119,19 +113,13 @@ verify-operator-bundle: operator-sdk
 	hack/verify-operator-bundle.sh
 
 operator-bundle:
-	@echo "FUSION=$(FUSION)"
```
Comment (malayparida2000) [inline Makefile:122]: Same, This change is not required.

File: Makefile [hunk 3]
```diff
@@ -119,19 +113,13 @@ verify-operator-bundle: operator-sdk
...
+	@echo "Building ocs-operator-bundle image"
 	hack/build-operator-bundle.sh
 
 operator-index:
-	@echo "FUSION=$(FUSION)"
-	@echo "Building ocs index image in sqlite db based format"
+	@echo "Building ocs-operator-index image"
```
Comment (malayparida2000) [inline Makefile:120]: Same

File: Makefile [hunk 4]
```diff
@@ -119,19 +113,13 @@ verify-operator-bundle: operator-sdk
...
 	hack/build-operator-index.sh
 
-operator-catalog:
-	@echo "FUSION=$(FUSION)"
-	@echo "Building ocs catalog image in file based catalog format"
```
Comment (malayparida2000) [inline Makefile:132]: Same

**Review by malayparida2000 — CHANGES_REQUESTED** [review]
I think we also need to remove the must gather iamge reference in
hack/gen-latest-csv.sh, generate-unified-csv.sh, tools/csv-merger.go.
@umangachapagain wdyt?

File: deploy/ocs-operator/manifests/ocs-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -3281,8 +3281,6 @@ spec:
     name: ceph-container
   - image: quay.io/csiaddons/k8s-sidecar:v0.5.0
     name: csiaddons-sidecar
-  - image: quay.io/ocs-dev/ocs-must-gather:latest
-    name: ocs-must-gather
```
Comment (umangachapagain) [inline deploy/ocs-operator/manifests/ocs-operator.clusterserviceversion.yaml:3285]: We should not remove this. `RelatedImages` is a list of images that we use, even if we do not maintain it in this repo.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: deploy/ocs-operator/manifests/ocs-operator.clusterserviceversion.yaml (hunk 5 above)
Comment (iamniting) [reply deploy/ocs-operator/manifests/ocs-operator.clusterserviceversion.yaml:3285]: +1

**Review by iamniting — CHANGES_REQUESTED** [review]
I believe that in order to be sorted with Malay comments, you must rebase makefile changes on top of main.

File: Makefile (hunk 2 above)
Comment (yati1998) [reply Makefile:122]: these changes occured after, make gen-csv, as mentioned in earlier CI failures. Cool I will rebase it.

**Review by umangachapagain — COMMENTED** [review]
Revert the changes in `hack/common.sh` and run `make gen-latest-csv`. Everything else is fine.

File: hack/common.sh [hunk 6]
```diff
@@ -76,24 +76,24 @@ OPERATOR_INDEX_NAME="${OPERATOR_INDEX_NAME:-${DEFAULT_OPERATOR_INDEX_NAME}}"
 FILE_BASED_CATALOG_NAME="${FILE_BASED_CATALOG_NAME:-${DEFAULT_FILE_BASED_CATALOG_NAME}}"
 CATALOG_IMAGE_NAME="${CATALOG_IMAGE_NAME:-${DEFAULT_CATALOG_IMAGE_NAME}}"
 METRICS_EXPORTER_IMAGE_NAME="${METRICS_EXPORTER_IMAGE_NAME:-${DEFAULT_METRICS_EXPORTER_IMAGE_NAME}}"
-MUST_GATHER_IMAGE_NAME="${MUST_GATHER_IMAGE_NAME:-${DEFAULT_MUST_GATHER_IMAGE_NAME}}"
 IMAGE_TAG="${IMAGE_TAG:-${DEFAULT_IMAGE_TAG}}"
+MUST_GATHER_IMAGE_NAME="${MUST_GATHER_IMAGE_NAME:-${DEFAULT_MUST_GATHER_IMAGE_NAME}}"
```
Comment (iamniting) [inline hack/common.sh:80]: I would suggest checking out this file from the commit as you are deleting and adding the same content at different lines.

**Review by iamniting — CHANGES_REQUESTED** [review]

**Comment by iamniting** [comment]
It appears that we are constructing the ocs must gather as part of the CI, which must be removed now in order to resolve the CI failures.

File: hack/common.sh (hunk 6 above)
Comment (malayparida2000) [reply hack/common.sh:80]: @iamniting as we will be no longer building the image from here. I think we should remove the variables needed for must gather & just put a direct link in single line as we do for Nooba, Rook, Ceph etc

**Comment by malayparida2000** [comment]
Yes, We need changes here 
https://github.com/openshift/release/blob/master/ci-operator/config/red-hat-storage/ocs-operator/red-hat-storage-ocs-operator-main.yaml.
We have to make changes here

> It appears that we are constructing the ocs must gather as part of the CI, which must be removed now in order to resolve the CI failures.

File: hack/common.sh (hunk 6 above)
Comment (iamniting) [reply hack/common.sh:80]: +1

Comment (yati1998) [reply hack/common.sh:80]: cool, going ahead with this then, I will remove the variables and directly add the must-gather image link

**Comment by malayparida2000** [comment]
@yati1998 , Do you want to make the openshift-ci changes yourself or should I do it?

**Comment by yati1998** [comment]
> @yati1998 , Do you want to make the openshift-ci changes yourself or should I do it?

Please feel free to add the changes @malayparida2000

File: hack/generate-latest-csv.sh [hunk 7]
```diff
@@ -14,7 +14,7 @@ export NOOBAA_CORE_IMAGE=${NOOBAA_CORE_IMAGE:-${LATEST_NOOBAA_CORE_IMAGE}}
 export NOOBAA_DB_IMAGE=${NOOBAA_DB_IMAGE:-${LATEST_NOOBAA_DB_IMAGE}}
 export CEPH_IMAGE=${CEPH_IMAGE:-${LATEST_CEPH_IMAGE}}
 export OCS_IMAGE=${OCS_IMAGE:-"${IMAGE_REGISTRY}/${REGISTRY_NAMESPACE}/${OPERATOR_IMAGE_NAME}:${IMAGE_TAG}"}
-export OCS_MUST_GATHER_IMAGE=${OCS_MUST_GATHER_IMAGE:-"${MUST_GATHER_FULL_IMAGE_NAME}"}
+export OCS_MUST_GATHER_IMAGE=${OCS_MUST_GATHER_IMAGE:-${LATEST_MUST_GATHER_IMAGE}}
```
Comment (iamniting) [inline hack/generate-latest-csv.sh:17]: You are also required to update the same in file `hack/dump-debug-info.sh`

**Review by iamniting — CHANGES_REQUESTED** [review]

File: hack/dump-debug-info.sh [hunk 8]
```diff
@@ -8,7 +8,7 @@ echo "dumping debug information"
 
 echo "Running ocs-must-gather to ${OCS_MUST_GATHER_DIR}"
 mkdir -p "$OCS_MUST_GATHER_DIR"
-${OCS_OC_PATH} adm must-gather --image "$MUST_GATHER_FULL_IMAGE_NAME" --dest-dir "$OCS_MUST_GATHER_DIR"
+${OCS_OC_PATH} adm must-gather --image "$OCS_MUST_GATHER_IMAGE" --dest-dir "$OCS_MUST_GATHER_DIR"
```
Comment (malayparida2000) [inline hack/dump-debug-info.sh:11]: The image here should be "$LATEST_MUST_GATHER_IMAGE", As this script is just sourcing from common.sh, And there it's defined as such.

**Comment by malayparida2000** [comment]
Rest everything looks good, Once the ci change PR here https://github.com/openshift/release/pull/36083 merges, The ci here can be retriggered and it should pass.

**Comment by yati1998** [comment]
@malayparida2000 can you please check why is this sanity check failing?

**Comment by malayparida2000** [comment]
@yati1998 , Ignore that sanity check. It has been little flaky in recent past, We will re run it few times & it will pass.

**Comment by iamniting** [comment]
> @yati1998 , Ignore that sanity check. It has been little flaky in recent past, We will re run it few times & it will pass.

It is passed after re-running.

**Comment by iamniting** [comment]
The changes look good to me. We are about to merge the CI PR once that is merged we will rerun the tests and add the LGTM.

**Review by iamniting — APPROVED** [review]

**Review by malayparida2000 — APPROVED** [review]

**Pull request merged**